tuna/db/tuna_tables.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+###############################################################################
+#
+# MIT License
+#
+# Copyright (c) 2022 Advanced Micro Devices, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+###############################################################################
+""" Module for creating DB tables"""
+import enum
+from sqlalchemy import Column, Integer, String, ForeignKey
+from sqlalchemy import Text, Enum
+from sqlalchemy.ext.declarative import declared_attr
+
+
+#pylint: disable=too-few-public-methods
+class JobEnum(enum.Enum):
+  """Represents job_enum column in config table"""
+  # pylint: disable=invalid-name ; names represent entries in job_enum column
+  # pylint: disable=duplicate-code
+  new = 1
+  started = 2
+  running = 3
+  completed = 4
+  timeout = 5
+  error_status = 6
+  no_update = 7
+  errored = 8
+  bad_param = 9
+  error = 10
+  transfer_error = 11
+  eval_start = 12
+  evaluating = 13
+  evaluated = 14
+  evaluate_error = 15
+  compile_start = 16
+  compiling = 17
+  compiled = 18
+  compile_error = 19
+  not_applicable = 20
+  aborted = 21
+  not_tunable = 22
+  compiled_pend = 23
+  evaluated_pend = 24
+
+
+class FinStep(enum.Enum):
+  """ Allowed Fin Steps """
+  # pylint: disable=invalid-name ; tuna/go_fish.py names valid fin steps as FinStep.__members__
+  find_compile = 1
+  find_eval = 2
+  get_solvers = 3
+  get_applicability = 4
+  not_fin = 5
+  miopen_find_compile = 6
+  miopen_find_eval = 7
+  miopen_perf_compile = 8
+  miopen_perf_eval = 9
```

Comment:
FinStep should not be present in this file. It is MIOpen specific. Leave this in miopen_tables.